paddle/phi/kernels/funcs/compare_functors.h

```diff
@@ -47,6 +47,38 @@ struct EqualFunctor {
   }
 };
 
+template <typename OutT>
+struct EqualFunctor<phi::dtype::complex<float>, OutT> {
+  HOSTDEVICE OutT operator()(const phi::dtype::complex<float>& a,
+                             const phi::dtype::complex<float>& b) const {
+    if (isinf(a.real) || isinf(a.imag) || isinf(b.real) || isinf(b.imag)) {
+      return a == b;
+    }
+    if (isnan(a.real) || isnan(a.imag) || isnan(b.real) || isnan(b.imag)) {
+      return false;
+    }
+    float epsilon = 1e-8f;
+    return std::abs(a.real - b.real) < epsilon &&
+           std::abs(a.imag - b.imag) < epsilon;
+  }
+};
+
+template <typename OutT>
+struct EqualFunctor<phi::dtype::complex<double>, OutT> {
+  HOSTDEVICE OutT operator()(const phi::dtype::complex<double>& a,
+                             const phi::dtype::complex<double>& b) const {
+    if (isinf(a.real) || isinf(a.imag) || isinf(b.real) || isinf(b.imag)) {
+      return a == b;
+    }
+    if (isnan(a.real) || isnan(a.imag) || isnan(b.real) || isnan(b.imag)) {
+      return false;
+    }
+    double epsilon = 1e-8;
+    return std::abs(a.real - b.real) < epsilon &&
+           std::abs(a.imag - b.imag) < epsilon;
+  }
+};
```

Comment:
这俩能合并为以下代码吗（可以参考: `Paddle/paddle/phi/kernels/funcs/matrix_inverse.h`）？
```suggestion
template <typename InT, typename OutT>
struct EqualFunctor<phi::dtype::complex<InT>, OutT> {
  HOSTDEVICE OutT operator()(const phi::dtype::complex<InT>& a,
                             const phi::dtype::complex<InT>& b) const {
    if (isinf(a.real) || isinf(a.imag) || isinf(b.real) || isinf(b.imag)) {
      return a == b;
    }
    if (isnan(a.real) || isnan(a.imag) || isnan(b.real) || isnan(b.imag)) {
      return false;
    }
    InT epsilon = 1e-8;
    return std::abs(a.real - b.real) < epsilon &&
           std::abs(a.imag - b.imag) < epsilon;
  }
};
```

Reply:
原来还能这么用